client.go

```diff
@@ -229,3 +238,19 @@ func (c *clientConfig) newSpec(t StreamType) Spec {
 		IsClient:   true,
 	}
 }
+
+func parseRequestURL(rawURL string) (*url.URL, *Error) {
```

Comment:
I moved this out of protocol.go and into client.go since this isn't protocol specific and was used exactly the same for each protocol. So I decided to change this simple to a `parseRequestURL` rather than `validateRequestURL` which seemed a bit more fitting for how it is used now and accomplishes the same goal.